### PR TITLE
make source host configurable #52

### PR DESCRIPTION
**Pull request by kimshinelove**
It worked subtracting the git domain in the configuration file. And adds the http authentication settings for the auth if necessary.

Can you apply the code?

### CODE REVIEW

File: lib/middleware/proxy-path.js [hunk 1]
```diff
@@ -40,6 +40,18 @@ module.exports = function (rootUrl) {
...
+        var requestOption = {
+            agent   : false,
+            hostname: parsedUrl.hostname,
+            path    : url.resolve(parsedUrl.path, pathWithIndex(req)),
+            headers : headers
+        };
+
+        // auth for private repository
+        // username:general_tokken
+        if(config.auth) {
```
Comment (rgrove) [inline lib/middleware/proxy-path.js]: I'm not interested in including this functionality in RawGit, but please feel free to maintain it in your fork.

File: web.js [hunk 2]
```diff
@@ -91,8 +91,8 @@ app.route('/:user/:repo/:branch/*')
         middleware.accessControl
     )
     .get(
-        middleware.fileRedirect('https://raw.githubusercontent.com'),
-        middleware.proxyPath('https://raw.githubusercontent.com')
+        middleware.fileRedirect(config.gitDomain || 'https://gist.githubusercontent.com'),
```
Comment (rgrove) [inline web.js]: This will result in GitHub repo files being requested from `gist.githubusercontent.com`, which won't work.

Comment (kimshinelove) [inline web.js]: sorry. it's my mistake.

i'll fix it.

File: lib/middleware/proxy-path.js (hunk 1 above)
Comment (kimshinelove) [inline lib/middleware/proxy-path.js]: I think I also think there is no association.
I'll manage to register another issue.

**Comment by rgrove** [comment]
Thanks! I cherry-picked the source host config changes and renamed the properties to more closely match my personal style: f58f556f

**Pull request closed without merging**